opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -480,18 +481,27 @@ private static void createGpsObservation(Event e, JSONObject jsonObject, String
             if (valueArr.length >= 2) {
                 String latitude = valueArr[0];
                 String longitude = valueArr[1];
-                addObs(e, formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.LATITUDE, AllConstants.TEXT, Collections.singletonList(latitude));
-                addObs(e, formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.LONGITUDE, AllConstants.TEXT, Collections.singletonList(longitude));
+                addGpsObs(e, formSubmissionFieldPrefix, AllConstants.GpsConstants.LATITUDE, latitude);
+                addGpsObs(e, formSubmissionFieldPrefix, AllConstants.GpsConstants.LONGITUDE, longitude);
                 if (valueArr.length >= 4) {
                     String altitude = valueArr[2];
                     String accuracy = valueArr[3];
-                    addObs(e, formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.ALTITUDE, AllConstants.TEXT, Collections.singletonList(altitude));
-                    addObs(e, formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.ACCURACY, AllConstants.TEXT, Collections.singletonList(accuracy));
+                    addGpsObs(e, formSubmissionFieldPrefix, AllConstants.GpsConstants.ALTITUDE, altitude);
+                    addGpsObs(e, formSubmissionFieldPrefix, AllConstants.GpsConstants.ACCURACY, accuracy);
                 }
             }
         }
```

Comment:
Why not use:

```
String[] gpsKeys = new String{AllConstants.GpsConstants.ALTITUDE,...,...};
for (int i = 0; i < valueArr; i++) {
   addGpsObs(e, formSubmissionFieldPrefix, gpsKeys[i], valueArr[i]);
}
```
This removes the need to check for lengths and is a bit more concise.